lib/honeybadger/config_sync_service.rb

```diff
@@ -0,0 +1,74 @@
+require 'honeybadger/check_in'
+module Honeybadger
+  class ConfigSyncService
+    def initialize(config)
+      @config = config
+      @check_ins_cache = nil
+    end
+
+    def sync_checkins
+      checkin_configs = @config.get(:checkins) || []
+      return [] if checkin_configs.empty?
+
+      checkins = checkin_configs.map do |cfg|
```

Comment:
I think there should be a check here whether there are any checkIns with the same names (under the same project id) and raise an error in that case.

Reply:
I looked at the laravel (or rather php) code for this and I don't see a similar check? If I'm missing it could you point to it?

Also, I wonder, if this is something that should be enforced if it would be much easier to enforce it server side?

(I agree that enforcing unique names makes sense if we want to allow people to use the name for running the check in)

Reply:
Unique names per project are enforced on the server side.

Reply:
@halfbyte [here](https://github.com/honeybadger-io/honeybadger-php/blob/f7d3a5902ce370e7fae22f084c5b9162f86035ab/src/CheckInsManager.php#L59) it is. You might have missed this because I only added the check last week.

> Unique names per project are enforced on the server side.

That's good! The issue for the client side is that whichever check-in comes last will overwrite the previous (at least that's what I experienced with PHP and JS) - so no error, just unexpected results.

Reply:
Oops - I lied... I was thinking of slugs. We don't currently require unique check-in names. I'll make that change. :)

Reply:
@stympy Are you making them unique globally or per project?
Because in our examples and documentation, we have check-in names such as _Hourly Notifications_. That's not a contestant for a globally unique name 😅. Besides, check-in ids are globally unique, so names could be unique per project, right?

Reply:
The uniqueness check for the name will be per-project.

Reply:
@halfbyte Are you planning to do implement this check here? I think it will help users with unexpected behavior. Consider the following:
- 2 _new_ checkins => `[ { name: 'my check-in', scheduleType: 'simple' }, { name: 'my check-in', scheduleType: 'cron' } ]`
- The sync command will go through the first check-in and call the backend to find out whether this is a new check-in or it already exists. Since it does not exist, it will issue a create new check-in request.
- For the second check-in, the backend (or the local in-memory cache) will return the same check-in and issue an update request.
- So, no error will be reported, but the check-ins defined in the config won't match what's on Honeybadger.

Reply:
@subzero10 I've added a simple check similar to the PHP one.